### PR TITLE
Plugin throwing exception if HTTP response code is not 200 has been fixed.

### LINKED ISSUE #139: Plugin throwing exception if HTTP responce code from elastic search is not 200
Catch this and show some proper message.

### PR DESCRIPTION
**Pull request by amit-shar**
fixes #139

### CODE REVIEW

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java [hunk 1]
```diff
@@ -415,6 +415,8 @@ public void mouseClicked(final MouseEvent e) {
         private final DefaultMutableTreeNode root;
         private final Notification notification;
         private Map<String, ArrayList<CodeInfo>> projectNodes;
+        private boolean flag;
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: We can make it volatile ?

Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: Also rename it to isFailed

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java [hunk 2]
```diff
@@ -431,18 +433,33 @@ public void mouseClicked(final MouseEvent e) {
...
-                doFrontEndWork(jTree, model, root, codePaneTinyEditorsInfoList, projectNodes);
-                jTabbedPane.setSelectedIndex(1);
+            if (!flag) {
+                if (!projectNodes.isEmpty()) {
+                    try {
+                        doFrontEndWork(jTree, model, root, codePaneTinyEditorsInfoList, projectNodes);
+                        jTabbedPane.setSelectedIndex(1);
+                    } catch (RuntimeException rte) {
+                        jTabbedPane.setSelectedIndex(0);
+                        showHelpInfo(rte.getMessage());
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: What if this also throws exception ? + showHelp will do setSelectedIndex(0) in it. 

I would go for rte.printstacktrace only.

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/ESUtils.java [hunk 3]
```diff
@@ -46,7 +45,8 @@
     private static final String TOKENS = "tokens";
     private static final String SOURCEFILE_SEARCH = "/sourcefile/_search?source=";
     private static final String REPOSITORY_SEARCH = "/repository/_search?source=";
-    private static final String FAILED_HTTP_ERROR_CODE = "Failed : HTTP error code : ";
+    private static final String FAILED_HTTP_ERROR_CODE = "<html>Failed : HTTP error code :" +
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/ESUtils.java]: Connection Error:

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/ESUtils.java [hunk 4]
```diff
@@ -131,8 +131,8 @@ public final String getESResultJson(final String esQueryJson, final String url)
 
             HttpResponse response = httpClient.execute(getRequest);
             if (response.getStatusLine().getStatusCode() != HTTP_OK_STATUS) {
-                throw new RuntimeException(FAILED_HTTP_ERROR_CODE + url
-                        + response.getStatusLine().getStatusCode());
+                throw new RuntimeException(FAILED_HTTP_ERROR_CODE +
+                        response.getStatusLine().getStatusCode());
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/ESUtils.java]: Put response.getResponseBody and error code both.

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java [hunk 5]
```diff
@@ -431,18 +433,33 @@ public void mouseClicked(final MouseEvent e) {
...
+                    }
+                } else {
+                    showHelpInfo(String.format(QUERY_HELP_MESSAGE, importsInLines.toString().replaceAll(",", "<br/>")));
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: Wrap this line.

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java (hunk 2 above)
Comment (amit-shar) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: Review feedback.

**Comment by ScrapCodes** [comment]
merged

**Pull request closed without merging**